liquidation_inspector.py

```diff
@@ -0,0 +1,105 @@
+# Internal imports
+from mev_inspect import utils
+from mev_inspect.config import load_config
+from mev_inspect.schemas.blocks import NestedTrace, TraceType
+from mev_inspect.schemas.classified_traces import Classification, ClassifiedTrace
+from mev_inspect.schemas.strategy import StrategyType, Strategy, Liquidation
+from mev_inspect.classifier_specs import CLASSIFIER_SPECS
+from mev_inspect.trace_classifier import TraceClassifier
+from mev_inspect import block
+
+# External Libraries
+import json
+import pandas as pd
+from typing import Optional
+from web3 import Web3
+from typing import List, Optional
+
+# TODO: adjust profit to new model
+#        unit test
+#        coinbase check / collateral source
+
+liquidations = []
+result = []
+
+# Protocol contract address must be in included, below is AaveLendingPoolCoreV1
+addrs = ['0x3dfd23A6c5E8BbcFc9581d2E864a68feb6a076d3']
+# Used to remove double-counted 'from' transfers
+from_doubles = []
+transfers_to = []
+transfers_from = []
+# Inspect list of classified traces and identify liquidation
+def find_liquidations(traces: List[ClassifiedTrace]):
+    tx = []
+
+    # For each trace
+    for count, trace in enumerate(traces):
+
+        # Check for liquidation and register trace and unique liquidator
+        if (
+            trace.classification == Classification.liquidate and
+            trace.transaction_hash not in tx
+            ):
+
+            liquidations.append(trace)
+            addrs.append(trace.from_address)
+            tx.append(trace.transaction_hash)
+
+            # Found liquidation, now parse inputs for data
+            for i in trace.inputs:
+
+                if(i == '_purchaseAmount'):
+                    liquidation_amount = trace.inputs[i]
+                elif (i == '_collateral'):
+                    collateral_type = trace.inputs[i]
+                    #This will be the address of the sent token
+                elif (i == '_reserve'):
+                    reserve = trace.inputs[i]
+                    #This will be the address of the received token
+                elif(i == '_user'):
+                    liquidated_usr = trace.inputs[i]
```

Comment:
these variables are each created inside an `if` branch which implies it's ok for them not to exist / they only are used inside of the `if`

but if the branch isn't hit for all of them the `Liquidation` creation will fail with a `No such variable found "liquidated_usr"` for example

Reply:
Also why `usr` not `user`?

One of my favorite articles in all of programming:
https://javascript.info/ninja-code

Take 5 mins and read it (ignore that it's about javascript - they're universal lessons)

Here's what they would have to say
![image](https://user-images.githubusercontent.com/5885679/129062234-adb5965e-fa35-4437-820e-49c711a80b5a.png)